### DISCUSSION TITLE
useWatch to call an external api

### DISCUSSION THREAD

**Discussion by T41WO** [react-hook-form/react-hook-form#3976] [Q&A] [answered]
I have two Controller wrapped  3rd-party UI dropdown components side by side. I am using useWatch to watch the selected value of dropdownA, and using this value to go fetch options for dropdownB. 

Is this the correct approach? I have an issue where the whole form is re-rendering after making the fetch call and causing the fetch call to be made over and over, ending in an infinite loop (and a "Too many re-renders" error).

**Comment by bluebill1049** [comment, marked as answer]
i think it's easier to just attach onChange or onBlur event on your input.

**Reply by T41WO** [reply]
Thanks. This solved the issue. Can you elaborate why useWatch caused the re-rendering?

**Reply by bluebill1049** [reply]
well, useWatch is actually made to isolate re-render, not skip re-render. eg

```
<form> // not render here
  <useWatch /> // we can isolate the re-render.
  ```
